src/server/routes/users.js

```diff
@@ -135,47 +138,98 @@ router.post('/create', adminAuthMiddleware('create a user.'), async (req, res) =
  * Route for updating user role
  */
 router.post('/edit', adminAuthMiddleware('update a user role'), async (req, res) => {
+	
 	const validParams = {
 		type: 'object',
-		required: ['users'],
+		required: ['user'],
 		properties: {
-			users: {
-				type: 'array',
-				items: {
-					type: 'object',
-					required: ['email', 'role'],
-					properties: {
-						email: {
-							type: 'string'
-						},
-						role: {
-							type: 'string',
-							enum: Object.values(User.role)
-						}
-
+			user: {
+				type: 'object',
+				required: ['id', 'email', 'role'],
+				properties: {
+					id: {
+						type: 'integer'
+					},
+					email: {
+						type: 'string'
+					},
+					role: {
+						type: 'string',
+						enum: Object.values(User.role)
+					},
+					password: {
+						type: 'string'
+					},
+					note: {
+						type: 'string'
 					}
 				}
 			}
 		}
 	};
+
 	if (!validate(req.body, validParams).valid) {
 		res.status(400).json({ message: 'Invalid params' });
 	} else {
+
 		try {
 			const conn = getConnection();
-			const { users } = req.body;
-			const minimumUser = users.find(user => user.role === User.role.ADMIN);
-			// This protects the database so that there will always be at least one admin during role updates.
-			if (minimumUser === undefined) {
-				res.sendStatus(400);
-			} else {
-				const roleUpdates = users.map(async user => await User.updateUserRole(user.email, user.role, conn));
-				await Promise.all(roleUpdates);
-				res.sendStatus(200);
+			const { user } = req.body;
+			const userBeforeChanges = await User.getByID(user.id,conn);
+			const numberOfAdmins = (await User.getNumberOfAdmins(conn)).count;
+			
+			// This protects the database so that there will always be at least one admin
+			if (numberOfAdmins < 2 && userBeforeChanges.role === 'admin' && user.role !== 'admin') {
+				const errorMessage = 'There must be at least one admin remaining to avoid lockout!';
+				log.error(errorMessage);
+				return res.status(400).json({
+					message: errorMessage,
+				});
+			}
+			
+			// set up Asynchronous database queries
+			const userUpdates = [];
+
+			// not the only admin, so proceed
```

Comment:
I'm okay with separating out every change but OED normally does an entire update so it is easier and one call. Now, here you cannot do the password if not provided with a new one so that could be a separate call. Up to you if you want to change this.

Reply:
Got it, it is now combined.